Sources/HttpPipeline/SignedCookies.swift

```diff
@@ -0,0 +1,149 @@
+import CommonCrypto
+import Cryptor
+import Foundation
+
+extension ResponseHeader {
+  /// Computes a signed cookie.
+  ///
+  /// - Parameters:
+  ///   - key: The name of the cookie.
+  ///   - value: The data of the cookie to sign.
+  ///   - options: (Optional) Extra options to attach to the cookie.
+  ///   - secret: The secret to sign the cookie with. This value must also be provided to verify a signature.
+  ///   - encrypt: (Optional) Further encrypts the signed cookie using the secret provided.
+  /// - Returns: A `Set-Cookie` header containing the signed cookie.
+  public static func setSignedCookie(
+    key: String,
+    data: Data,
+    options: Set<CookieOption> = [],
+    secret: String,
+    encrypt: Bool = false
+    )
+    -> ResponseHeader? {
+
+      let encodedValue = data.base64EncodedString()
+      guard let computedDigest = digest(value: encodedValue, secret: secret) else { return nil }
+
+      let signedValue = "\(encodedValue)--\(computedDigest)"
+      guard let finalValue = encrypt ? encrypted(text: signedValue, secret: secret) : signedValue
+        else { return nil }
+
+      return .some(.setCookie(key: key, value: finalValue, options: options.union([.httpOnly])))
```

Comment:
@stephencelis so I originally had all signed/encrypted cookies be of type `HttpOnly` and `Secure`, but the latter one kinda complicates the local dev story. it seems like it's ok to transport this cookie over non-https since it's already encrypted, and I can't find any information online. whatdya think?